drivers/console/native_posix_console.c

```diff
@@ -27,9 +39,199 @@ static int printk_init(struct device *arg)
 
 	extern void __printk_hook_install(int (*fn)(int));
 	__printk_hook_install(putchar);
+}
+
+/**
+ * Ensure that whatever was written thru printk is displayed now
+ */
+void posix_flush_stdout(void)
+{
+	fflush(stdout);
+}
+#endif /* CONFIG_NATIVE_POSIX_STDOUT_CONSOLE */
+
+#if defined(CONFIG_NATIVE_POSIX_STDIN_CONSOLE)
+
+static struct k_fifo *avail_queue;
+static struct k_fifo *lines_queue;
+static u8_t (*completion_cb)(char *line, u8_t len);
+
+static K_THREAD_STACK_DEFINE(stack, CONFIG_ARCH_POSIX_RECOMMENDED_STACK_SIZE);
+static struct k_thread native_stdio_thread;
+
+
+static inline void found_eof(void)
+{
+	/*
+	 * Once stdin is closed
+	 * or the input file has ended, there is no need to
+	 * try again
+	 */
+	ECHO("Got EOF\n");
+	k_thread_abort(&native_stdio_thread);
+}
+
+static int handle_native_shell_cmd(const char *s, u32_t *towait)
+{
+
+	if (strncmp(s, "!wait", 5) == 0) {
+		int ret;
+
+		ret = sscanf(&s[5], "%u", towait);
+		if (ret != 1) {
+			posix_print_error_and_exit("%s(): %s not understood, "
+					"syntax = !wait %%u\n",
+					__func__, s);
```

Comment:
The alignment of the split lines doesn't look quite right (could be the GH user interface also distorting stuff)

Reply:
Should it be aligned with the parenthesis or 1 tab indented farther than the function call?